pkg/deprecated/ecdsa.go

```diff
@@ -0,0 +1,109 @@
+package deprecated
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/theupdateframework/go-tuf/data"
+	"github.com/theupdateframework/go-tuf/pkg/keys"
+)
+
+func init() {
```

Comment:
Having access to the functions without the type maps is very little use: most people want to be importing this to use in the Repo or Client logic, where all keys are pulled under the hood through these maps.

Reply:
Ack for testing. That makes sense I think, but I'm again not sure if it'll be used outside the tests in this package already

Reply:
IMO it's worth it for testing alone. This would make it so that the tests didn't have to implement fake verifiers; I think that gives us more confidence in the cross-compatibility of the old/new versions.

Further, I think "importing this thing changes the behavior everywhere" is pretty surprising, even if it's documented, and so naming it something that makes it clearer that it's a destructive operation is a good idea.

Reply:
Separated out the init and the package code itself: users can implement tests with a deprecated ecdsa verifier, without importing the key into the map.